docs/architecture/authentication-and-authorization.md

```diff
@@ -0,0 +1,63 @@
+# Authentication and Authorization
+
+## Introduction
+Currently, communication between the Compass and both runtimes and applications are not secured. We need to provide some security possibilities.
+We want to secure the Compass using ORY's Hydra and OathKeeper. There would be two ways of authentication:
+ - Oauth 2.0 
+ - SSL/TLS Certificates.
+
+## Proposed solution
+As mentioned, there would be two possibilities for securing the connection. To achieve that, first, we need to integrate Hydra and OathKeeper into the Compass. Then we will introduce a new component - Tenant Service.
```

Comment:
Too many details in one sentence. I would go with something like this:

As mentioned, there would be two possibilities for securing the connection. To achieve that, first, we need to integrate Hydra and OathKeeper into the Compass. We need to implement additional supporting components to make our scenarios valid. On the Director component, we will implement an OathKeeper hydrator handler that will resolve client_id (which can be runtime ID or application ID) to particular tenant ID.

Reply:
Also, link the hydrator to some external document describing its concept.